lib/spack/llnl/util/filesystem.py

```diff
@@ -1240,6 +1240,44 @@ def get_single_file(directory):
     return fnames[0]
 
 
+@system_path_filter
+def compute_sfn(path: os.PathLike):
```

Comment:
I think `windows_sfn` might be a better name (I think it's worth indicating this is a Windows-specific action in the function name, and the term "compute" doesn't say much).

Reply:
Makes sense, will refactor.